src/canvas/SelectableCanvas.ts

```diff
@@ -1296,11 +1272,7 @@ export class SelectableCanvas<EventSpec extends CanvasEvents = CanvasEvents>
   _realizeGroupTransformOnObject(
     instance: FabricObject
   ): Partial<typeof instance> {
-    if (
-      instance.group &&
-      instance.group === this._activeSelection &&
-      this._activeObject === instance.group
-    ) {
+    if (instance.group && isActiveSelection(instance.group)) {
```

Comment:
this check is different now, we still have to check that instance.group is activeObject
Being an activeSelection, doesn't mean you are the active one.